homeassistant/components/purpleair/config_flow.py

```diff
@@ -232,6 +267,44 @@ async def async_step_by_coordinates(
 
         return await self.async_step_choose_sensor()
 
+    async def async_step_by_sensor_index(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the input of a sensor index."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="by_sensor_index", data_schema=SENSOR_INDEX_SCHEMA
+            )
+
+        sensor_index = int(user_input[CONF_SENSOR_INDEX])
```

Comment:
This would not work (it would raise) if a user provided a non-number input e.g. `a` or something